frontend/src/lib/WebsocketConnection.tsx

```diff
@@ -219,6 +219,10 @@ export class WebsocketConnection {
     return undefined
   }
 
+  public disconnect(): void {
+    this.setFsmState(ConnectionState.DISCONNECTED_FOREVER)
```

Comment:
I'm probably missing something, but how does this result in a connected websocket being disconnected? It looks like we'll cancel a connection _attempt_ inside `setFsmState`, but I don't see where we close the connection?

Reply:
`cancelConnectionAttempt` also closes any active websocket connections.

I think the potential confusion implies that we should rename the method to `closeConnection` since it seems reasonable to expect a `closeConnection` method to also cancel any in-flight connection attempts, but I wouldn't expect a `cancelConnectionAttempt` method to also close already-established connections

Reply:
Ah! Makes sense. Yeah we should change that name!